### PR TITLE
Added a Product Form with basic validation

### PR DESCRIPTION
**Pull request by merecka**
Added a Product Form with basic validation for the admin to add additional products to the app.

### CODE REVIEW

File: src/app/admin/product-form/product-form.component.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      ngModel
+      name="category"
+      id="category"
+      class="form-control"
+      required
+    >
+      <option value=""></option>
+      <option
+        *ngFor="let category of categoriesObservable | async"
+        [value]="category.key"
```
Comment (BarretteBanner) [inline src/app/admin/product-form/product-form.component.html:50]: My code has category.$key. But if it works without the $ it should be fine.

Comment (ahaly92) [reply src/app/admin/product-form/product-form.component.html:50]: yea does this work without the $

**Review by ahaly92 — APPROVED** [review]
if it works ship it

File: src/app/admin/product-form/product-form.component.html (hunk 1 above)
Comment (merecka) [reply src/app/admin/product-form/product-form.component.html:50]: Everything seems to function properly with it so far.

**Pull request merged**